overlord/snapstate/handlers.go

```diff
@@ -783,6 +784,30 @@ func (m *SnapManager) undoMountSnap(t *state.Task, _ *tomb.Tomb) error {
 	return m.backend.RemoveSnapDir(snapsup.placeInfo(), otherInstances)
 }
 
+// queryDisabledServices returns a list of all currently disabled snap services
+// in the snap
+// note this function takes a snap info rather than snapst because there are
+// situations where we want to call this on non-current snap infos, i.e. in the
+// undo handlers, see undoLinkSnap for an example
+func (m *SnapManager) queryDisabledServices(info *snap.Info, pb progress.Meter) ([]string, error) {
+	// save the list of services that are in the disabled state before unlinking
+	// and thus removing the snap services
+	snapSvcStates, err := m.backend.ServicesEnableState(info, pb)
+	if err != nil {
+		return nil, err
+	}
+
+	disabledSnapSvcs := []string{}
+	// add all disabled services to the list
+	for svc, isEnabled := range snapSvcStates {
+		if !isEnabled {
+			disabledSnapSvcs = append(disabledSnapSvcs, svc)
+		}
+	}
+
+	return disabledSnapSvcs, nil
```

Comment:
Should disabledSnapSvcs be sorted to ease testing?

Reply:
Due to your previous suggestion about renaming one thing, a number of tests I am adding are made more difficult because this isn't sorted, so you were right this is a good idea and I will implement this :smile:

Reply:
This is done